src/Couchbase.Lite.Shared/API/Sync/ReplicatorProgress.cs

```diff
@@ -71,14 +71,14 @@ public struct ReplicatedDocument
         [CanBeNull]
         public CouchbaseException Error { get; }
 
-        internal Exception Exception { get; set; }
+        internal CouchbaseLiteException Exception { get; set; }
```

Comment:
Still this is not helpful because it's **internal**.  The public facing `Error` is where you want to record this information so that the user can actually see it.  The `Error` should be as follows:

1. Original exception is CouchbaseException (or derived) -> Set `Error` directly
2. Original exception is not CouchbaseException (or derived) -> Create a new `CouchbaseLiteException` with the original exception as `InnerException` and set `Error` to that
3. No original exception, just error code -> Use `CouchbaseException.Create()`